pep-0458.txt

```diff
@@ -509,11 +509,43 @@ discarded and more offline keys will not help resist key recovery attacks [21]_
 unless diversity of keys is maintained.
 
 
-Online and Offline Keys Recommended for Each Role
--------------------------------------------------
+Managing online keys
+---------------------
 
-In order to support continuous delivery, the *timestamp*, *snapshot*, *bins*
-role keys MUST be online.
+In order to support continuous delivery, the keys for the *timestamp*,
+*snapshot*, all *bin-n* roles MUST be online. There is little benefit in
+requiring all of these roles to use different online keys, since attackers
+would presumably be able to compromise all of them if they compromise PyPI.
+Therefore, it is reasonable to use one online key for them all.
```

Comment:
I just saw that this needs to be adopted in the [preceding paragraph](https://github.com/secure-systems-lab/peps/blob/master/pep-0458.txt#L476-L479):
> Different keys for online roles allow for each of the keys to be placed on separate servers if need be, and prevents side channel attacks that compromise one key from automatically compromising the rest of the keys.

Reply:
Good catch, thanks!